### PR TITLE
Task/283 task replace bluebird lib with p limit

### PR DESCRIPTION
**Pull request by DougMidgley**
# PR details

## What is the purpose of this pull request? (put an "X" next to an item)

_Please delete options that are not relevant._

- [ ] Documentation update
- [ ] Bug fix
- [ ] New metadata support
- [ ] Enhanced metadata
- [ ] Add a CLI option
- [ ] Add something to the core
- [X] Other, please explain: Moved out of bluebird promises to reduce overhead - closes #284 

## What changes did you make? (Give an overview)

Removed Bluebird promises(used for concurrency only) and replaced with P-limit

## Is there anything you'd like reviewers to focus on?

Ensure Tests PR is done first. I tried testing retrieve on Assets and also deploy of query, but would be good if you can double check

## Checklist

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] ESLint & Prettier are not outputting errors or warnings
- [X] README.md updated (if applicable)
- [] CHANGELOG.md updated

### CODE REVIEW

File: lib/metadataTypes/MetadataType.js [hunk 1]
```diff
@@ -354,18 +352,24 @@ class MetadataType {
                 Util.metadataLogger('error', this.definition.type, 'upsert', ex, metadataKey);
             }
         }
+        const createLimit = pLimit(10);
```
Comment (JoernBerkefeld) [inline lib/metadataTypes/MetadataType.js:356]: what is this doing?

File: lib/metadataTypes/MetadataType.js [hunk 2]
```diff
@@ -354,18 +352,24 @@ class MetadataType {
...
-                { concurrency: 10 }
+            await Promise.all(
+                metadataToCreate
+                    .filter((r) => r !== undefined && r !== null)
+                    .map((metadataEntry) =>
+                        createLimit(() => this.create(metadataEntry, deployDir))
+                    )
             )
         ).filter((r) => r !== undefined && r !== null);
+        const updateLimit = pLimit(10);
```
Comment (JoernBerkefeld) [inline lib/metadataTypes/MetadataType.js:366]: same as above

File: package.json [hunk 3]
```diff
@@ -45,7 +45,7 @@
         "inquirer": "8.2.2",
         "json-to-table": "4.2.1",
         "mustache": "4.2.0",
-        "p-limit": "4.0.0",
+        "p-limit": "3.1.0",
```
Comment (JoernBerkefeld) [inline package.json:48]: why an older version?

**Review by JoernBerkefeld — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (DougMidgley) [reply package.json:48]: same logic, but version 4 uses ESM modules, which was causing issues when running our or CJS context. 
Suggestion would be to hold until a full move to ESM modules OR TS

File: lib/metadataTypes/MetadataType.js (hunk 1 above)
Comment (DougMidgley) [reply lib/metadataTypes/MetadataType.js:356]: This is essentially the rate limiter object, which is applied to each of the iterables. In this case it is limiting it to 10 promises at any given time

File: lib/metadataTypes/MetadataType.js (hunk 2 above)
Comment (DougMidgley) [reply lib/metadataTypes/MetadataType.js:366]: cf. other comment

**Pull request merged**